packages/openneuro-server/src/graphql/permissions.js

```diff
@@ -78,31 +83,39 @@ export const checkDatasetRead = (datasetId, userId, userInfo) => {
  * @param {object} state Level to verify
  */
 
-export const checkDatasetWrite = (
+export const checkDatasetWrite = async (
   datasetId,
   userId,
   userInfo,
   state = states.WRITE,
+  checkExists = true,
```

Comment:
Since the signature is changing anyways, maybe we should make this an options object argument? That way the signature can remain the same in the future. Not sure, either way seems pretty good here.